Source/DafnyCore/AST/Expressions.cs

```diff
@@ -1035,7 +1035,13 @@ public DatatypeValue(IToken tok, string datatypeName, string memberName, List<Ex
   }
 
   public override IEnumerable<Expression> SubExpressions {
-    get { return Arguments; }
+    get {
+      // Arguments is available only when fully resolved. Otherwise,
+      // only unresolved arguments (Actual) are available.
+      return Arguments ??
+             Bindings.ArgumentBindings?.Select(binding => binding.Actual) ??
+             Enumerable.Empty<Expression>();
+    }
```

Comment:
```suggestion
      return Arguments ?? Enumerable.Empty<Expression>();
```

Reply:
There is no reason that we should return null or an empty list for sub-expressions when there _are_ sub-expressions in the AST.
Are you ok if I fix the comment that you mention?

Reply:
Moreover, in the upcoming formatter PR, there will be a new field "PreResolveSubExpressions", which is mostly equal to SubExpressions. But SubExpressions' behavior is undefined before Resolution, so it's up to us to decide what its meaning is. And what more obvious to the meaning that sub-expressions to be rexolved? WDYT?

Reply:
I think we shouldn't change that comment without changing the implementation of every other SubExpressions field, and I think that's beyond the scope of this PR, no?  And I don't think we should introduce new inconsistencies.